psl/psl-core/src/validate/validation_pipeline/validations/relation_indexes.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{datamodel_connector::RelationMode, validate::validation_pipeline::context::Context};
+use diagnostics::DatamodelWarning;
+use parser_database::ast::WithSpan;
+use parser_database::walkers::RelationFieldWalker;
+
+// { x_1, x_2, ..., x_n } is left-wise included in { y_1, y_2, ..., y_m } if and only if
```

Comment:
Cool math. But, maybe write a proper rustdoc comment with `///` where you explain exactly what this does in proper english?

Reply:
Also private functions go under the public ones, is usually how we do these.

Reply:
So I should move that fn definition down? Alright